### PR TITLE
fix: org admin delete other teams

### LINKED ISSUE #11205: [CAL-2416] Org owner/Admin can't delete any sub-team
### Issue Summary

Owner/admin can't delete any sub-team, whether is a team they belong or any other.

### Steps to Reproduce

1. Create an org
2. Create a sub-team
3. Try to delete created sub-team

### Actual Results

Don't have the option to delete sub-team

### Expected Results

Be able to delete sub-team

<sub>[CAL-2416](https://linear.app/calcom/issue/CAL-2416/org-owneradmin-cant-delete-any-sub-team)</sub>

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Adds functionality so owner/admin in orgs can delete teams they don't belong to.
- Adds code to other team listing so disband can appear as an option
- Add trpc function to org router so admin/owners can deleteTeams

Fixes #11205 

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/features/ee/organizations/pages/components/OtherTeamList.tsx [hunk 1]
```diff
@@ -24,10 +24,9 @@ export default function OtherTeamList(props: Props) {
     }
   }
 
-  const deleteTeamMutation = trpc.viewer.teams.delete.useMutation({
+  const deleteTeamMutation = trpc.viewer.organizations.deleteTeam.useMutation({
```
Comment (alannnc) [inline packages/features/ee/organizations/pages/components/OtherTeamList.tsx:27]: Now using different method for orgs

File: packages/features/ee/organizations/pages/components/OtherTeamList.tsx [hunk 2]
```diff
@@ -24,10 +24,9 @@ export default function OtherTeamList(props: Props) {
...
     async onSuccess() {
-      await utils.viewer.teams.list.invalidate();
-      await utils.viewer.teams.hasTeamPlan.invalidate();
+      await utils.viewer.organizations.listOtherTeams.invalidate();
```
Comment (alannnc) [inline packages/features/ee/organizations/pages/components/OtherTeamList.tsx:29]: invalidating correct listing

File: packages/features/ee/organizations/pages/components/OtherTeamListItem.tsx [hunk 3]
```diff
@@ -119,6 +122,32 @@ export default function OtherTeamListItem(props: Props) {
...
+                        title={t("disband_team")}
+                        confirmBtnText={t("confirm_disband_team")}
+                        isLoading={props.isLoading}
+                        onConfirm={() => {
+                          props.onActionSelect("disband");
+                        }}>
+                        {t("disband_team_confirmation_message")}
+                      </ConfirmationDialogContent>
+                    </Dialog>
+                  </DropdownMenuItem>
```
Comment (alannnc) [inline packages/features/ee/organizations/pages/components/OtherTeamListItem.tsx:150]: Add Disband option

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
Screenshot 2023-09-11 at 5 13 52 PM

Getting 401 when trying to disband team from /settings/organizations/teams/other/[team_id]/profile

File: packages/trpc/server/routers/viewer/organizations/deleteTeam.handler.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      teamId: input.teamId,
+    },
+  });
+
+  await prisma.team.delete({
+    where: {
+      id: input.teamId,
+    },
+  });
+};
```
Comment (Udit-takkar) [inline packages/trpc/server/routers/viewer/organizations/deleteTeam.handler.ts:22]: i think we should use prisma.$transaction here

**Comment by leog** [comment]
@alannnc was the feedback from @Udit-takkar tackled?

**Comment by alannnc** [comment]
> @alannnc was the feedback from @Udit-takkar tackled?
@leog 
Changes were addressed. Good call @Udit-takkar

**Review by Udit-takkar — APPROVED** [review]
LGTM

**Pull request merged**